pkg/controller/cloud/node_controller.go

```diff
@@ -349,21 +355,18 @@ func (cnc *CloudNodeController) AddCloudNode(obj interface{}) {
 
 		nodeAddresses, err := getNodeAddressesByProviderIDOrName(instances, curNode)
 		if err != nil {
-			glog.Errorf("%v", err)
-			return nil
+			return err
```

Comment:
If getting node addresses fails we should return an error instead of just returning nil here. If a provider wants to skip node addresses they can return `[]v1.NodeAddresses{}, nil`.

Reply:
+1